app/lib/update_global_activity_analytics.rb

```diff
@@ -51,61 +21,106 @@ def perform
+      json_key_io: File.open("config/secure/service_account_cred.json"),
+      scope: "https://www.googleapis.com/auth/analytics.readonly"
+    )
+
+    # Authorize with our readonly credentials
+    service.authorization = credentials
+
+    @google_client = service
   end
 
-  def prepare_query
-    today = Time.zone.today
-    metrics = FIELDS.map.with_index { |(_, column), index|
-      attrs = {
-        columnId: index.to_s,
-        id: column[:name]
-      }
-      column.fetch(:options, {}).each { |key, val| attrs[key] = val }
-      attrs
-    }
-
-    date_range = format(
-      DATE_RANGE_TEMPLATE,
-      start_year: today.year,
-      end_year: (today + 1.year).year
+  def date_range
+    Google::Apis::AnalyticsreportingV4::DateRange.new(
+      start_date: Date.today.beginning_of_year.to_s,
+      end_date: (Date.today.beginning_of_year + 1.year).to_s
     )
+  end
+
+  def sessions_and_users_request
+    metrics = [Google::Apis::AnalyticsreportingV4::Metric.new(
+      expression: "ga:users"
+    ),
+
+      Google::Apis::AnalyticsreportingV4::Metric.new(
+        expression: "ga:sessions",
+        alias: "launches"
+      )]
 
-    query = BASE_QUERY.clone
-    query["globalFilters"] = [
-      {
-        "type": "dateRange",
-        "dateRange": date_range
-      }
-    ]
-    query["metricContainer"] = {"metrics": metrics}
-    query
+    Google::Apis::AnalyticsreportingV4::ReportRequest.new(
+      view_id: ENV.fetch("GOOGLE_ANALYTICS_VIEW_ID"),
+      sampling_level: "DEFAULT",
+      date_ranges: [date_range],
+      metrics: metrics,
+      dimensions: []
+    )
   end
 
-  def fetch_counters(data)
-    Hash[FIELDS.keys.zip(data["summaryData"]["totals"].map(&:to_i))]
+  def gospel_presentations_request
+    metrics = [Google::Apis::AnalyticsreportingV4::Metric.new(
+      expression: "ga:totalEvents",
+      alias: "gospel_presentations"
+    )]
+
+    Google::Apis::AnalyticsreportingV4::ReportRequest.new(
+      view_id: ENV.fetch("GOOGLE_ANALYTICS_VIEW_ID"),
+      sampling_level: "DEFAULT",
+      filters_expression: "ga:eventLabel==Presenting the Gospel",
+      date_ranges: [date_range],
+      metrics: metrics,
+      dimensions: []
+    )
+  end
+
+  def countries_request
+    metrics = [Google::Apis::AnalyticsreportingV4::Metric.new(
+      expression: "ga:sessions",
+      alias: "countries"
+    )]
+
+    countries_dimesion = Google::Apis::AnalyticsreportingV4::Dimension.new(
+      name: "ga:country"
+    )
+
+    Google::Apis::AnalyticsreportingV4::ReportRequest.new(
+      view_id: ENV.fetch("GOOGLE_ANALYTICS_VIEW_ID"),
+      sampling_level: "DEFAULT",
+      date_ranges: [date_range],
+      metrics: metrics,
+      dimensions: [countries_dimesion]
+    )
+  end
+
+  def google_analytics_report
+    # Create a new report request
+    request = Google::Apis::AnalyticsreportingV4::GetReportsRequest.new(
+      {report_requests: [sessions_and_users_request, gospel_presentations_request, countries_request]}
```

Comment:
I'm surprised we need the `{}` here.
Does this work?
```suggestion
      report_requests: [sessions_and_users_request, gospel_presentations_request, countries_request]
```